File: Project-Euler/Stats/euler76/euler76.py
import sys

target = int(sys.argv[1])

def calc(target):
  ways = [1 for i in xrange(target + 2)]
  for i in xrange(2, target):
    for j in xrange(i, target + 1):
      ways[j + 1] += ways[j + 1 - i]
  return ways[-1]

print calc(target)

File: Project-Euler/Stats/euler76/euler76.java
import java.util.*;

public class euler76
{
	public static void main(String[] args)
	{
		int target = Integer.parseInt(args[0]);
		System.out.println(calc(target));
	}
	
	private static int calc(int target)
	{
		List<Integer> ways = new ArrayList<>();
		for(int i = 0; i <= target + 1; i++)
		{
			ways.add(1);
		}
		for(int i = 2; i < target; i++)
		{
			for(int j = i; j <= target; j++)
			{
				ways.set(j + 1, ways.get(j + 1) + ways.get(j + 1 - i));
			}
		}
		return ways.get(ways.size() - 1);
	}
}
